.ci/Jenkinsfile

```diff
@@ -86,23 +88,27 @@ pipeline {
         stages {
           stage('build'){
             steps {
-              withGithubNotify(context: "Build-${PLATFORM}") {
-                deleteDir()
-                unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
-                withMageEnv(){
-                  dir("${BASE_DIR}"){
-                    cmd(label: 'Go build', script: 'mage build')
+              stageStatusCache(id: "Build-${PLATFORM}", bucket: env.JOB_GCS_BUCKET, credentialsId: env.JOB_GCS_EXT_CREDENTIALS) {
+                withGithubNotify(context: "Build-${PLATFORM}") {
+                  deleteDir()
```

Comment:
Isn't this line to L96 redundant because of L178-L181? And if so, would it be similar for the other invocations of stageStatusCache()?

Reply:
I'm not sure if I understand your question correctly. 
These changes did nothing but add the top level stageStatusCache closure, just one line change and then the indentation for the inner closure.

For instance, the given snippet:

```
withGithubNotify(context: "Build-${PLATFORM}") {
  deleteDir()
  unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
  withMageEnv(){
    dir("${BASE_DIR}"){
```

was transformed to:


```
stageStatusCache(id: "Build-${PLATFORM}", bucket: env.JOB_GCS_BUCKET, credentialsId: env.JOB_GCS_EXT_CREDENTIALS) {
  withGithubNotify(context: "Build-${PLATFORM}") {
    deleteDir()
    unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
    withMageEnv(){
      dir("${BASE_DIR}"){
```

What do you mean with the redundant?

Reply:
I mean that, for example, deletedir() is being called twice and I believe that's not necessary. In your example above, it's being called inside first as a result of the stageStatusCache and then directly inside of withGithubNotify.

Perhaps I'm not understanding the intended behavior here though? We can hop on a quick call and go over it and maybe you can show me if it's not clear from the above.